business/mesh.go

```diff
@@ -334,7 +334,7 @@ func findKialiInNamespace(ctx context.Context, namespace string, clusterName str
 				services = kubernetes.FilterServicesByLabels(selector, tmpSvc)
 			}
 		} else {
-			services, getSvcErr = layer.k8s.GetServicesByLabels(kialiNs.Name, "app.kubernetes.io/part-of=kiali")
+			services, getSvcErr = layer.k8sClients[clusterName].GetServicesByLabels(kialiNs.Name, "app.kubernetes.io/part-of=kiali")
```

Comment:
Not sure if we would need to add a k8sClients in MeshService. At the moment, it is created using the HomeCluster k8s. But, as findKialiInNamespace is passing the layer as parameter, layer needs the k8sClients - So that is why this change.